src/Npgsql/NpgsqlParameter.cs

```diff
@@ -20,21 +20,21 @@ public class NpgsqlParameter : DbParameter, IDbDataParameter, ICloneable
     {
         #region Fields and Properties
 
-        byte _precision;
-        byte _scale;
-        int _size;
+        private protected byte _precision;
```

Comment:
I think it would be slightly better design to expose a protected constructor and to keep the fields private

Reply:
It's `private` already with visibility to the only one child in the same assembly.